be/src/olap/olap_cond.cpp

```diff
@@ -637,7 +637,7 @@ bool Conditions::delete_conditions_eval(const RowCursor& row) const {
     }
     
     for (auto& each_cond : _columns) {
-        if (each_cond.second->is_key() && !each_cond.second->eval(row)) {
+        if (_cond_column_is_key_or_duplicate(each_cond.second) && !each_cond.second->eval(row)) {
```

Comment:
I think check agg type is more clear?

Reply:
agg type has two may be null or AGG_NONE, it maybe  more complicated